src/MOLFiniteDifference/MOL_discretization.jl

```diff
@@ -117,13 +117,20 @@ function SciMLBase.symbolic_discretize(pdesys::ModelingToolkit.PDESystem,discret
             grid_indices = CartesianIndices(((axes(g)[1] for g in grid)...,))
             depvarsdisc = map(depvars) do u
                 if t == nothing
-                    [Num(Variable{Real}(Base.nameof(operation(u)),II.I...)) for II in grid_indices]
+                    @show "here1"
+                    sym = nameof(operation(u))
+                    collect(first(@variables $sym[collect(axes(g)[1] for g in grid)...]))
                 elseif isequal(arguments(u),[t])
+                    @show "here2"
                     [u for II in grid_indices]
                 else
-                    [Num(Variable{Symbolics.FnType{Tuple{Any}, Real}}(Base.nameof(operation(u)),II.I...))(t) for II in grid_indices]
+                    sym = nameof(operation(u))
+                    @show collect(axes(g)[1] for g in grid)
+                    @show @variables $sym[collect(axes(g)[1] for g in grid)...](t)
+                    collect(first(@variables $sym[collect(axes(g)[1] for g in grid)...](t)))
                 end
```

Comment:
```julia
Base.OneTo{Int64}[Base.OneTo(21), Base.OneTo(11)]
#= C:\Users\accou\.julia\dev\DiffEqOperators\src\MOLFiniteDifference\MOL_discretization.jl:129 =# @variables((($(Expr(:$, :sym)))[collect(((axes(g))[1] for g = grid))...])(t)) = Symbolics.Arr{Num, 1}[(map(Symbolics.CallWith((t,)), u))[Base.OneTo(21)]]
depvarsdisc = Vector{Num}[[u[1, 1](t), u[2, 1](t), u[3, 1](t), u[4, 1](t), u[5, 1](t), u[6, 1](t), u[7, 1](t), u[8, 1](t), u[9, 1](t), u[10, 1](t), u[11, 1](t), u[12, 1](t), u[13, 1](t), u[14, 1](t), u[15, 1](t), u[16, 1](t), u[17, 1](t), u[18, 1](t), u[19, 1](t), u[20, 1](t), u[21, 1](t)]]
```

@YingboMa @shashi let's talk about this one today. I think this might be a bug in `@variables` parsing, but `@variables u[1:21,1:11]` works...

Reply:
```julia
@show @variables $sym[collect(axes(g)[1] for g in grid)](t)
```

does

```julia
#= C:\Users\accou\.julia\dev\DiffEqOperators\src\MOLFiniteDifference\MOL_discretization.jl:129 =# @variables((($(Expr(:$, :sym)))[collect(((axes(g))[1] for g = grid))])(t)) = Symbolics.Arr{Num, 1}[(map(Symbolics.CallWith((t,)), u))[Base.OneTo{Int64}[Base.OneTo(21), Base.OneTo(11)]]]
```

so yeah, it's dropping the second index for some reason.